### PR TITLE
Bug 1210301 - Intermittent utility_tray_gestures r=mhenretty

### PR DESCRIPTION
**Pull request by albertopq**

### CODE REVIEW

**Comment by autolander** [comment]
Autolander could not find a bug number in your pull request title. All pull requests should be in the format of: Bug [number] - [description].

**Comment by autolander** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=e6e93e21b5858c6bc93fe9cfb38f3e99fa6562af

**Comment by autolander** [comment]
Autolander could not find a bug number in your pull request title. All pull requests should be in the format of: Bug [number] - [description].

**Comment by autolander** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=1a533eb307f1b9656246b1fb47bf258920dd59ca

**Comment by autolander** [comment]
Autolander could not find a bug number in your pull request title. All pull requests should be in the format of: Bug [number] - [description].

**Comment by autolander** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=3248cec260174bbc0b1be629ca9b40578785f025

**Comment by autolander** [comment]
Autolander could not find a bug number in your pull request title. All pull requests should be in the format of: Bug [number] - [description].

**Comment by autolander** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=c943f5cb47e5acdd1752c66b11724856cbb93afc

**Comment by autolander** [comment]
Autolander could not find a bug number in your pull request title. All pull requests should be in the format of: Bug [number] - [description].

**Comment by autolander** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=2b64779fcad20da1baf084a7efb9b73f5c66e088

**Comment by autolander** [comment]
Autolander could not find a bug number in your pull request title. All pull requests should be in the format of: Bug [number] - [description].

**Comment by autolander** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=4a2a0ef89269a5bfb5c0f7115f0c914a243d4265

**Comment by autolander** [comment]
Autolander could not find a bug number in your pull request title. All pull requests should be in the format of: Bug [number] - [description].

**Comment by autolander** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=c6c73b79f0afc4f970f9fa103568b5631ed6d372

**Comment by autolander** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=c6c73b79f0afc4f970f9fa103568b5631ed6d372

File: apps/system/test/marionette/lib/utility_tray.js [hunk 1]
```diff
@@ -64,7 +64,13 @@
...
 
     get quickSettings() {
-      return this.client.helper.waitForElement(this.Selectors.quickSettings);
+      var element;
+      this.client.waitFor(function() {
+        element = this.client.findElement(this.Selectors.quickSettings);
+        return element && element.displayed();
+      }.bind(this));
+
+      return element;
```
Comment (mikehenrty) [inline apps/system/test/marionette/lib/utility_tray.js]: `waitForElement` basically does the above, but has more robust logic for when an element is not accessible [1]. What advantage does this change have?

1.) https://github.com/mozilla-b2g/gaia/blob/2a6e31cdf1dabad180ff9dbc3eddce440d00ca5e/tests/jsmarionette/plugins/marionette-helper/index.js#L147

**Comment by autolander** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=db42cc670c599e7bf6d9fb49ee30102e9cf478c3

File: apps/system/test/marionette/lib/utility_tray.js (hunk 1 above)
Comment (albertopq) [inline apps/system/test/marionette/lib/utility_tray.js]: I saw https://bugzilla.mozilla.org/show_bug.cgi?id=1228753 error in some the runs I triggered, but didn't know that waitForElement was already checking if is visible, so let's keep it as it was. Thanks!

**Comment by autolander** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=1a572e047b4766292130d100309df2f239fd768d

**Comment by autolander** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=b774d035716fa38065d22738c322d7d94a5c2965

**Comment by autolander** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=778afe5b4cbeea244b151bc9639331907a31e7c1

**Comment by autolander** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=8f0be280e615fbfe68a73ab09c2f93f1d6e6bdec

**Pull request merged**